mini-projects/00-jq-rs/src/main.rs

```diff
@@ -1,3 +1,99 @@
+use anyhow::Result;
+use clap::Parser;
+use colored::Color;
+use serde_json::Value;
+use std::fs::File;
+use std::io::{self, BufRead, BufReader, Lines};
+use std::path::PathBuf;
+
+//mod input;
+mod filters;
+mod printers;
+
+// - `color-output` (True)
+// - `monochrome-output` (False)
+// - `sort-keys` (False)
+// - `indent` (2) - this value must be in the range of 0-7 inclusive
+// - `compact-output` (False)
+
+#[derive(Parser, Debug)]
+struct Args {
+    #[clap(short, long)]
+    color_output: Option<bool>,
+
+    #[clap(short = 'v', long)]
+    compact_output: Option<bool>,
+
+    #[clap(short, long)]
+    monochrome_output: Option<bool>,
+
+    #[clap(short, long)]
+    sort_keys: Option<bool>,
+
+    #[clap(short, long)]
+    indent: Option<usize>,
+
+    filter_string: String,
+
+    json_file: Option<PathBuf>,
+}
+
 fn main() {
     println!("Hello World!");
+    let args = Args::parse();
+    //println!("{:?}", args);
+
+    // println!("{:?}", args);
+    //let input_vec = input::read_general_input(args.file)?;
+
+    //parse filter string
+    //let mut filters: Vec<Filter> = Vec::new();
+    // let filters = parse_filters
+
+    //test_valid_args(args)
+    let file_path = args.json_file.as_ref().expect("JSON file path is required");
+
+    let file = File::open(file_path).unwrap();
+    let reader = BufReader::new(file);
+    let json_obj: serde_json::Value = serde_json::from_reader(reader).unwrap();
+
+    // if let Some(array) = json_obj.as_array() {
+
+    // }
+
+    let filters = match filters::parse_filter_sequence(&args.filter_string) {
+        Ok(f) => f,
+        Err(e) => {
+            eprintln!("Error parsing filter: {}", e);
+            std::process::exit(1);
+        }
+    };
+
+    // Apply the filters
+    let initial_values = vec![json_obj];
```

Comment:
I see you're wrapping the json in a vec to make the array iterator interface more generic later - I see what you're doing, but it *may* have been better to just have the value within apply_filters to be an enum of either a actual SerdeJson or an Array of SerdeJsons, that would make it more explicit what the distinction is